### PR TITLE
add mail sending service

### PR DESCRIPTION
**Pull request by SergiLomtadze**

### CODE REVIEW

File: src/ExadelMentorship.BusinessLogic/ExadelMentorship.BusinessLogic.csproj [hunk 1]
```diff
@@ -4,9 +4,10 @@
     <TargetFramework>netstandard2.0</TargetFramework>
 	 <LangVersion>9</LangVersion>
   </PropertyGroup>
-
+	
   <ItemGroup>
     <PackageReference Include="FluentValidation" Version="11.0.1" />
+    <PackageReference Include="MailKit" Version="3.3.0" />
```
Comment (kdavletov-exadel) [inline src/ExadelMentorship.BusinessLogic/ExadelMentorship.BusinessLogic.csproj:10]: You can remove it

File: src/ExadelMentorship.BusinessLogic/Services/Mail/EmailSender.cs [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+                client.Send(mailMessage);
+            }
+            catch (Exception ex)
+            {
+                throw new AuthenticationException(ex.Message);
+            }
+            finally
+            {
+                client.Disconnect(true);
```
Comment (kdavletov-exadel) [inline src/ExadelMentorship.BusinessLogic/Services/Mail/EmailSender.cs:52]: `SmtpClient.Dispose` might already do this (and more). So you may need this finally block.

**Review by kdavletov-exadel — CHANGES_REQUESTED** [review]

File: src/ExadelMentorship.BusinessLogic/Services/Mail/EmailSender.cs [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+                client.Send(mailMessage);
+            }
+            catch (Exception ex)
+            {
+                throw new AuthenticationException(ex.Message);
+            }
+            finally
+            {
+                client.Dispose();
```
Comment (kdavletov-exadel) [inline src/ExadelMentorship.BusinessLogic/Services/Mail/EmailSender.cs:52]: https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/using-statement

**Review by kdavletov-exadel — CHANGES_REQUESTED** [review]

File: src/ExadelMentorship.BusinessLogic/Services/Mail/EmailSender.cs (hunk 2 above)
Comment (SergiLomtadze) [reply src/ExadelMentorship.BusinessLogic/Services/Mail/EmailSender.cs]: done

File: src/ExadelMentorship.BusinessLogic/ExadelMentorship.BusinessLogic.csproj (hunk 1 above)
Comment (SergiLomtadze) [reply src/ExadelMentorship.BusinessLogic/ExadelMentorship.BusinessLogic.csproj:10]: I can not remove it, I'm using it SmtpClient, 
If I will use 'SmtpClient' from System.Net.Mail then it does not have 'Connect, Authenticate etc'
image

Comment (SergiLomtadze) [reply src/ExadelMentorship.BusinessLogic/ExadelMentorship.BusinessLogic.csproj:10]: maybe it's better to stay with MailKit? :-) 
image

**Review by kdavletov-exadel — CHANGES_REQUESTED** [review]

File: src/ExadelMentorship.BusinessLogic/ExadelMentorship.BusinessLogic.csproj (hunk 1 above)
Comment (SergiLomtadze) [reply src/ExadelMentorship.BusinessLogic/ExadelMentorship.BusinessLogic.csproj]: done

**Comment by SergiLomtadze** [comment]
done

**Review by kdavletov-exadel — APPROVED** [review]

**Pull request merged**